### DISCUSSION TITLE
Ping info

### DISCUSSION THREAD

**Discussion by NovaliX-Dev** [lichess-org/mobile#284] [Ideas]
For the moment we only have the ping icon when playing games.

# The idea
When we will click on this icon, a small box with the ping we have and the server latency is shown.

It should dissipear after a short amout of time, or when we click on the icon again.

Maybe we can color the text depending on the ping (Green if everything is fine, yellow if it's less fine and red if it's very slow), but i guess it will make the code more complex.

Here is some designs i've sketched : 
## In game #1 : 
![image](https://github.com/lichess-org/mobile/assets/79244938/6c6a46bf-3fe2-48a2-a8bb-6e456f44267e)

Pros : 
- More consistent design with the box in the lobby

Cons : 
- It's on the chessboard (only on tablets in landscape). Definitively not a great place. Maybe put the icon on the other side will solve the issue ?

## In game #2 : 
![image](https://github.com/lichess-org/mobile/assets/79244938/5af1662c-9530-4d12-99f3-017da50d7460)

Pros : 
- Don't overlap with the chessboard

Cons : 
- Design not consistent anymore
- More complex to implement, as it depends on the platform the app runs on 

## Lobby #1
![image](https://github.com/lichess-org/mobile/assets/79244938/8b8fc664-3d84-456a-aa1a-4e0233758ae7)

# Why
We have this informations on the website and on the previous lichess app. Moreover, it can help us make diagnostics if we have problems connecting to lichess.

I'm hearing for feedback about this idea and the designs, so don't hesitate to share yours.

**Comment by veloce** [comment]
Thanks for sharing ideas. 

For technical reasons, only idea number 1, in the game screen will be possible.

We can't show the ping info on home because we need an open websocket connection to get ping info. Contrary to the previous app, I don't open a permanent websocket because: 1- it is useless in most of the screen, 2- it drains the battery faster.

It's better with a tooltip because ping and server lag numbers are non-essential infos and it's fine to hide them behind a click. Showing them permanently would just clutter the UI.

**Comment by NovaliX-Dev** [comment]
> We can't show the ping info on home because we need an open websocket connection to get ping info. Contrary to the previous app, I don't open a permanent websocket because: 1- it is useless in most of the screen, 2- it drains the battery faster.

What about this : when we click on the icon, we send a request to the lichess server to get the server latency ? That way, we can wait for the response with the server and compute the ping based on the time it took only when we want to. No permanent websocket open that way.

I'm not sure this is possible with lichess' api though.

Also i didn't meant to add it on all screens, just in the lobby and in the game, where it's the most logical.

> It's better with a tooltip because ping and server lag numbers are non-essential infos and it's fine to hide them behind a click. Showing them permanently would just clutter the UI.

That's essentialy the idea i had in mind : only showing them when needed and make them disapear after a short amount of time, or when we click on the icon again.

Can you show me how a tooltip looks ?

**Reply by veloce** [reply]
> What about this : when we click on the icon, we send a request to the lichess server to get the server latency ? That way, we can wait for the response with the server and compute the ping based on the time it took only when we want to. No permanent websocket open that way.

It's not really possible because the icon itself need the websocket connection (to show green/yellow/red/reconnecting states) and it takes a few seconds to get the ping when we open a websocket.

Have a look at flutter doc for the tooltip: https://api.flutter.dev/flutter/material/Tooltip-class.html

**Reply by NovaliX-Dev** [reply]
> Have a look at flutter doc for the tooltip: https://api.flutter.dev/flutter/material/Tooltip-class.html

Looks great !

> It's not really possible because the icon itself need the websocket connection (to show green/yellow/red/reconnecting states) and it takes a few seconds to get the ping when we open a websocket.

I've done some research on periodic requests vs websockets and on mobile websockets seems more efficient. So doing periodic GET on lichess from the mobile will drain the battery even more.

Of course this depends of the time between each request, but i guess that will be too long to be even practical.

So the ping icon on the lobby seems not to be a good idea from the battery perspective overall (at least the way i saw it).
